collectors/python.d.plugin/hpssa/hpssa.chart.py

```diff
@@ -0,0 +1,402 @@
+                    break
+
+        adapter['logical_drives'].append(ld)
+
+    def _parse_physical_drive(self, adapter, fqn):
+        pd = {
+            'fqn': fqn,
+            'status': None,
+            'temperature': None,
+        }
+
+        for line in self:
+            if _adapter_regex.match(line) or \
+                    _array_regex.match(line) or \
+                    _drive_regex.search(line) or \
+                    _ignored_sections_regex.match(line):
+                self._rewind()
+                break
+
+            match = _key_value_regex.match(line)
+            if match:
+                key, value = match.group('key', 'value')
+                if key == 'Status':
+                    pd['status'] = value == 'OK'
+                elif key == 'Current Temperature (C)':
+                    pd['temperature'] = int(value)
+            else:
+                self._rewind()
+                break
+
+        adapter['physical_drives'].append(pd)
+
+    def _parse_ignored_section(self):
+        for line in self:
+            if _adapter_regex.match(line) \
+                    or _array_regex.match(line) \
+                    or _drive_regex.match(line) \
+                    or _ignored_sections_regex.match(line) \
+                    or not _key_value_regex.match(line):
+                self._rewind()
+                break
+
+
+class Service(ExecutableService):
+    def __init__(self, configuration=None, name=None):
+        super(Service, self).__init__(configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = deepcopy(CHARTS)
+        self.ssacli_path = self.configuration.get('ssacli_path', 'ssacli')
+        self.use_sudo = self.configuration.get('use_sudo', True)
+        self.cmd = []
+
+    def _get_adapters(self):
+        try:
+            adapters = HPSSA(self._get_raw_data(command=self.cmd)).adapters
+            if not adapters:
+                # If no adapters are returned, run the command again but capture stderr
+                err = self._get_raw_data(command=self.cmd, stderr=True)
+                if err:
+                    raise HPSSAException('Error executing cmd {}: {}'.format(' '.join(self.cmd), '\n'.join(err)))
+            return adapters
+        except HPSSAException as ex:
+            self.error(ex)
+            return []
+
+    def check(self):
+        if not os.path.isfile(self.ssacli_path):
+            ssacli_path = find_binary(self.ssacli_path)
+            if ssacli_path:
+                self.ssacli_path = ssacli_path
+            else:
+                self.error('Cannot locate "{}" binary'.format(self.ssacli_path))
+                return False
+
+        if self.use_sudo:
+            sudo = find_binary('sudo')
+            if not sudo:
+                self.error('Cannot locate "{}" binary'.format('sudo'))
+                return False
+
+            err = self._get_raw_data(command=[sudo, '-n', '-v'], stderr=True)
```

Comment:
i suggest to use `-n -l`, see

> https://github.com/netdata/netdata/issues/7107

Reply:
Even better than the solution in #7107 is that `sudo -l` accepts a command and checks the user is allowed to run that:

```
                   If a command is specified and is permitted by the security
                   policy, the fully-qualified path to the command is
                   displayed along with any command line arguments.  If
                   command is specified but not allowed, sudo will exit with a
                   status value of 1.
```